CMakeLists.txt

```diff
@@ -0,0 +1,209 @@
+### setup project ###
+# https://numpy.org/doc/stable/f2py/buildtools/skbuild.html
+cmake_minimum_required(VERSION 3.18)
+
+project(${SKBUILD_PROJECT_NAME}
+  VERSION ${SKBUILD_PROJECT_VERSION}
+  DESCRIPTION "Utilities for reading WRF output"
+  LANGUAGES C Fortran
+  )
+
+# Safety net
+if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
+  message(
+    FATAL_ERROR
+      "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there.\n"
+  )
+endif()
+
+find_package(Python 3.8 REQUIRED COMPONENTS Development.Module NumPy)
+
+# Ensure scikit-build modules
+if (NOT SKBUILD)
+  # Kanged --> https://github.com/Kitware/torch_liberator/blob/master/CMakeLists.txt
+  # If skbuild is not the driver; include its utilities in CMAKE_MODULE_PATH
+  execute_process(
+    COMMAND "${Python_EXECUTABLE}"
+    -c "import os, skbuild; print(os.path.dirname(skbuild.__file__))"
+    OUTPUT_VARIABLE SKBLD_DIR
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  list(APPEND CMAKE_MODULE_PATH "${SKBLD_DIR}/resources/cmake")
+  message(STATUS "Looking in ${SKBLD_DIR}/resources/cmake for CMake modules")
+endif()
+
+# Grab the variables from a local Python installation
+# NumPy headers
+# F2PY headers
+execute_process(
+  COMMAND "${Python_EXECUTABLE}"
+  -c "import numpy.f2py; print(numpy.f2py.get_include())"
+  OUTPUT_VARIABLE F2PY_INCLUDE_DIR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+find_package(OpenMP COMPONENTS Fortran)
+set_source_files_properties(fortran/ompgen.F90 PROPERTIES Fortran_PREPROCESS ON)
+# TODO: Figure out the conditionals for running the C Preprocessor on Fortran files
+# I think the main thing to be changed is -E -cpp
+# Intel is -fpp -save-temps or /fpp on Windows
+# or call fpp instead of the fortran compiler to get it to stop after preprocessing
```

Comment:
It'd be good to look into this for Windows at least, but I think that can happen after getting this merged.  Maybe along with some testing for Windows.